packages/stores/src/queries/concentrated-liquidity/liquidity-net-in-direction.ts

```diff
@@ -0,0 +1,133 @@
+import { KVStore } from "@keplr-wallet/common";
+import {
+  ChainGetter,
+  ObservableChainQuery,
+  ObservableChainQueryMap,
+} from "@keplr-wallet/stores";
+import { Dec, Int } from "@keplr-wallet/unit";
+import { LiquidityDepth } from "@osmosis-labs/math";
+import {
+  ConcentratedLiquidityPool,
+  TickDataProvider,
+} from "@osmosis-labs/pools";
+import { computed, makeObservable } from "mobx";
+
+import { LiquidityNetInDirection } from "./types";
+
+const BOUND_TICK = 3420;
+
+type QueryStoreParams = {
+  poolId: string;
+  tokenInDenom: string;
+};
+
+/** Stores tick data for a single pool swapping in a single direction by token in. */
+export class ObservableQueryLiquidityNetInDirection extends ObservableChainQuery<LiquidityNetInDirection> {
+  @computed
+  get depthsInDirection(): LiquidityDepth[] {
+    return (
+      this.response?.data.liquidity_depths.map((depth) => {
+        return {
+          tickIndex: new Int(depth.tick_index),
+          netLiquidity: new Dec(depth.liquidity_net),
+        };
+      }) ?? []
+    );
+  }
+
+  @computed
+  get currentTick() {
+    return new Int(this.response?.data.current_tick ?? 0);
+  }
```

Comment:
No opposite direction. It will return this same reference until any observable in `this.response.data.current_tick` changes.

References are a problem when passed to memoization primitives such as useMemo/use* deps and computedFn params.